pkg/snowflake/parameters.go

```diff
@@ -683,52 +683,119 @@ func GetParameterDefault(key string) ParameterDefault {
+	stmt := fmt.Sprintf("ALTER ACCOUNT SET %s = %s", v.key, v.value)
+	_, err := v.db.Exec(stmt)
+	return err
+}
+
+// SessionParameterBuilder abstracts the creation of SQL queries for Snowflake session parameters.
+type SessionParameterBuilder struct {
+	key       string
+	value     string
+	onAccount bool
+	user      string
+	db        *sql.DB
+}
+
+func NewSessionParameter(key, value string, db *sql.DB) *SessionParameterBuilder {
+	return &SessionParameterBuilder{
+		key:   key,
+		value: value,
+		db:    db,
+	}
+}
+
+func (v *SessionParameterBuilder) SetOnAccount(onAccount bool) *SessionParameterBuilder {
+	v.onAccount = onAccount
+	return v
+}
+
+func (v *SessionParameterBuilder) SetUser(user string) *SessionParameterBuilder {
+	v.user = user
+	return v
+}
+
+func (v *SessionParameterBuilder) SetParameter() error {
+	if v.onAccount {
+		// prepared statements do not work here for some reason. We already validate inputs so its okay
+		stmt := fmt.Sprintf("ALTER ACCOUNT SET %s = %s", v.key, v.value)
+		_, err := v.db.Exec(stmt)
+		return err
+	}
+	if v.user == "" {
+		return fmt.Errorf("user is required when setting session parameters on a user")
+	}
+	stmt := fmt.Sprintf("ALTER USER %s SET %s = %s", v.user, v.key, v.value)
+	_, err := v.db.Exec(stmt)
+	return err
+}
+
+// ObjectParameterBuilder abstracts the creation of SQL queries for Snowflake object parameters.
+type ObjectParameterBuilder struct {
 	key              string
 	value            string
-	parameterType    ParameterType
+	onAccount        bool
 	objectType       ObjectType
 	objectIdentifier string
 	db               *sql.DB
 }
 
-func NewParameter(key, value string, parameterType ParameterType, db *sql.DB) *ParameterBuilder {
-	return &ParameterBuilder{
-		key:           key,
-		value:         value,
-		parameterType: parameterType,
-		db:            db,
+func NewObjectParameter(key, value string, db *sql.DB) *ObjectParameterBuilder {
+	return &ObjectParameterBuilder{
+		key:   key,
+		value: value,
+		db:    db,
 	}
 }
 
-func (v *ParameterBuilder) WithObjectType(objectType ObjectType) *ParameterBuilder {
+func (v *ObjectParameterBuilder) SetOnAccount(onAccount bool) *ObjectParameterBuilder {
+	v.onAccount = onAccount
+	return v
+}
+
+func (v *ObjectParameterBuilder) WithObjectType(objectType ObjectType) *ObjectParameterBuilder {
 	v.objectType = objectType
 	return v
 }
 
-func (v *ParameterBuilder) WithObjectIdentifier(objectIdentifier string) *ParameterBuilder {
+func (v *ObjectParameterBuilder) WithObjectIdentifier(objectIdentifier string) *ObjectParameterBuilder {
 	v.objectIdentifier = objectIdentifier
 	return v
 }
 
-func (v *ParameterBuilder) SetParameter() error {
-	// Should this be set on the account level?
-	setOnAccount := v.parameterType == ParameterTypeAccount || v.parameterType == ParameterTypeSession || v.parameterType == ParameterTypeObject && v.objectIdentifier == ""
-	if setOnAccount {
+func (v *ObjectParameterBuilder) SetParameter() error {
```

Comment:
There's a lot of code duplication between this method and `SessionParameterBuilder.SetParameter`, couldn't the latter be implemented as something like:
```go
o := NewObjectParameter(v.key, v.value)
o.WithObjectType(USER)
o.SetOnAccount(v.onAccount)
o.SetParameter()
```